homeassistant/components/hive/climate.py

```diff
@@ -177,8 +191,10 @@ def set_preset_mode(self, preset_mode) -> None:
 
             self.session.heating.turn_boost_on(self.node_id, 30, temperature)
 
-        for entity in self.session.entities:
-            entity.handle_update(self.data_updatesource)
+    async def async_added_to_hass(self):
+        """When entity is added to Home Assistant."""
+        await super().async_added_to_hass()
+        self.session.entity_lookup.update({self.entity_id: self.node_id})
```

Comment:
```py
self.session.entity_lookup[self.entity_id] = self.node_id
```

Reply:
We need to connect the signal, `DOMAIN`, to a callback update method here. I'm thinking we then need this in all platform entity classes, so a common base class would be good.

Reply:
I have updated how it is added to the dict, but i am not sure what you mean by connect the signal and domain to a call back method.

Reply:
This part:
https://github.com/home-assistant/home-assistant/blob/0e157856027d854e4183e00d91c830bfcdad877f/homeassistant/components/iaqualink/__init__.py#L192-L198